### PR TITLE
UT for BPF-mode host endpoint policy

### PR DESCRIPTION
**Pull request by neiljerram**
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by tomastigera — APPROVED** [review]
LGTM, strong testing! :+1: Some questions, I think mostly for my benefit.

File: dataplane/linux/bpf_ep_mgr_test.go
Comment (neiljerram) [reply dataplane/linux/bpf_ep_mgr_test.go:208]: Agree that's a bit confusing.  I've moved the genPolicy call to the Contexts that need it.

File: dataplane/linux/bpf_ep_mgr_test.go
Comment (neiljerram) [reply dataplane/linux/bpf_ep_mgr_test.go:278]: No, because empty tiers/profiles for normal policy is not the same as being completely skipped or suppressed.  (In fact empty tiers/profiles = Deny.)  So there's no reason for the calling code to zero out the tiers/profiles fields when it sets SuppressNormalHostPolicy = true.

**Review by fasaxc — APPROVED** [review]
One nit

File: dataplane/linux/bpf_ep_mgr_test.go
Comment (neiljerram) [reply dataplane/linux/bpf_ep_mgr_test.go:244]: I'm afraid it is async because ensureProgramAttached and updatePolicyProgram are called from within goroutines.  So, as you say, we should have locks...

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Pull request merged**
